### ISSUE TITLE
How do we make our charts more accessible if possible?

### ISSUE THREAD

**Issue by SaptakS** [freedomofpress/pressfreedomtracker.us#1628]
I have few concerns regarding the accessibility of the charts. Firstly, I think the way [Vertical bar Charts](https://github.com/freedomofpress/pressfreedomtracker.us/pull/1614) is really much more accessible compared to what we have for our homepage, since we have a separate field for the description of the svg chart. That helps the screenreader users to get an idea about the visualization. I think we should have a similar `<desc>` element in homepage with dynamic content (since it's a little bit more dynamic, but if someone has idea for static content, I am fine with that).

But there are few other points that bugs me, mainly because our charts are interactive:
- The homepage charts are interactive in 2 ways - the chart itself, filters based on years and tags. 
  - The year and tag filters are actually keyboard navigable and actually update the chart. This is good for keyboard users (and similar AT users) since they can see the updated result, but not so much for screenreader users. The screenreader users problem would be solved if we had dynami `<desc>` for each of the charts.
  - The chart itselves work only with mouse and touch right now. That's bad because we are ignoring almost all other AT users, since they can't really navigate to the incident database page using the same way a mouse and touch user can.
- I think for all the other charts as well, they are only interactive with mouse and touch, and not interactive at all for any AT users, including keyboard
- If we do plan to make them interactive for keyboard and screen reader users, we might not be able to get away with just a `<desc>` and each of those interactive components should have accessible name/description
- For a visualization like we have in home page, if we do make it interactive with keyboard, I think we should add a "Skip visualization" link so a user can skip the visualization instead of having to tab through the entire visualization.

These are some of the points that come to my mind. There can obviously be more discussion points. I feel at the very least, we should definitely have a `<desc>` for the homepage visualization, even if it's a static one.

**Comment by harrislapiroff** [comment]
I have [one example](https://observablehq.com/@harrislapiroff/us-press-freedom-tracker-visualizations#map) of an accessible chart I attempted from the pre accurat days (specifically the US map on that page and not the bar chart). There were two specific things I did for this:

1. Used actual `<a>` link elements for the interactive components and
2. Added aria roles and labels to the marks on the chart in order to get a screenreader to read them in a human-friendly way

I bet we could achieve these with React Animated Dateset, but we may have to coerce it a bit or fork it.

**Comment by harrislapiroff** [comment]
Part of the work here is being done in #1651. Let's keep this ticket open for ongoing assessment of how we're doing here.

**Comment by SaptakS** [comment]
In the accessibility audit, it was also noticed that some of the chart colors don't meet the color contrast requirements

<img width="660" height="500" alt="Image" src="https://github.com/user-attachments/assets/bb57ed66-2388-42bc-b09b-794299fd8b09" />

**Comment by SaptakS** [comment]
Another SC failure from Accessibility Audit: the popup shown on hovering in the map only shows on mouse over and there is no way of activating or dismissing it via keyboard.

<img width="660" height="240" alt="Image" src="https://github.com/user-attachments/assets/6f0c6daa-8d6e-4705-8a10-1af1044889ce" />

**Comment by SaptakS** [comment]
Another issue mentioned in the Accessibility Audit related to the map is that some of the circles dont meet the[ minimum target size SC](https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html). They are either very small or very close to each other or overlapping, that makes clicking on one of them difficult.
